src/main/java/task/Deadline.java

```diff
@@ -0,0 +1,87 @@
+package task;
+import errors.DukeException;
+import java.time.format.DateTimeParseException;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDate;
+
+public class Deadline extends Task {
+    
+    public static final String DATE_STRING_PATTERN = "MMM dd yyyy";
+    private LocalDate by;
+
+    public Deadline(String description, LocalDate by) {
+        super(description);
+        this.type = "D";
+        this.by = by;
+    }
+
+    public LocalDate getBy() {
+        return this.by;
+    }
+
+    public void setBy(LocalDate by) {
+
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return super.toString() + " (by: " + by.format(DateTimeFormatter.ofPattern(DATE_STRING_PATTERN)) + ")";
+    }
+
+    /**
+     * Get task name of task type deadline
+     *
+     * @param taskDetails string to retrieve task name from
+     *
+     */
+    public static String getDeadlineTask(String taskDetails) {
+        //first space
+        int firstSpaceIndex = taskDetails.indexOf(" ");
+        int byIndex = taskDetails.indexOf("/by");
+        String deadlineTask = taskDetails.substring(firstSpaceIndex+1,byIndex-1).trim();
+        if (deadlineTask.length() == 0){
+            throw new StringIndexOutOfBoundsException();
+        }
+        return deadlineTask.trim();
+    }
+
+    /**
+     * Based on user input generate and return a date object
+     *
+     * @param dateString user input of date string
+     *
+     */
+    public static LocalDate getDateObject(String dateString){
```

Comment:
Can consider making putting this function in Task since it will also be used by event